common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -4,26 +4,32 @@
 from django_filters.rest_framework import DjangoFilterBackend
 from edx_rest_framework_extensions.authentication import JwtAuthentication
 from rest_framework import permissions, viewsets
-from rest_framework.authentication import SessionAuthentication
 
 from entitlements.api.v1.filters import CourseEntitlementFilter
-from entitlements.models import CourseEntitlement
+from entitlements.api.v1.permissions import IsAdminOrAuthenticatedReadOnly
 from entitlements.api.v1.serializers import CourseEntitlementSerializer
+from entitlements.models import CourseEntitlement
+from openedx.core.djangoapps.cors_csrf.authentication import SessionAuthenticationCrossDomainCsrf
 from student.models import CourseEnrollment
 
 log = logging.getLogger(__name__)
 
 
 class EntitlementViewSet(viewsets.ModelViewSet):
-    authentication_classes = (JwtAuthentication, SessionAuthentication,)
-    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser,)
-    queryset = CourseEntitlement.objects.all().select_related('user')
+    authentication_classes = (JwtAuthentication, SessionAuthenticationCrossDomainCsrf,)
+    permission_classes = (permissions.IsAuthenticated, IsAdminOrAuthenticatedReadOnly,)
     lookup_value_regex = '[0-9a-f-]+'
     lookup_field = 'uuid'
     serializer_class = CourseEntitlementSerializer
     filter_backends = (DjangoFilterBackend,)
     filter_class = CourseEntitlementFilter
 
+    def get_queryset(self):
```

Comment:
do we want the ability for a staff user to pass a user param and get the entitlements for that user?

Reply:
Do we have a need for that right now? I think we should add it if the need arises.  I am refactoring this section right now to only allow the admin to get their list of entitlements.

Reply:
Yes we have this using URL parameter of user=username

Reply:
You should use a filter to keep this code simple. Otherwise, you run the risk of introducing errors and confusion down the road as this API evolves. An explicit API call allows the client to know exactly what data to expect and ensures that the same URL returns the same set of data regardless of who makes the request (assuming the requester has the appropriate permissions). See the guidelines at https://openedx.atlassian.net/wiki/spaces/OpenDev/pages/18350757/edX+REST+API+Conventions#edXRESTAPIConventions-ExplicitFilters.